### PR TITLE
Consent or Pay: Guardian Light Landing Page route

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

- Setup a UK-only page `/uk/guardian-light` landing page
- Route  other regions to `/contributions` landing page (specified by your geoId)
- Setup a UK-only page `/guardian-light` route to either above (dependent on geoId)

[**Trello Card**](https://trello.com)

## How to test

|https://support.thegulocal.com/uk/guardian-light|https://support.thegulocal.com/us/guardian-light|
|-----|-----|
|image|image|

### CODE REVIEW

File: support-frontend/conf/routes [hunk 1]
```diff
@@ -64,6 +64,15 @@ GET  /$countryGroupId<(uk|us|au|eu|int|nz|ca)>/checkout            controllers.A
 GET  /$countryGroupId<(uk|us|au|eu|int|nz|ca)>/one-time-checkout   controllers.Application.productCheckoutRouter(countryGroupId: String)
 GET  /$countryGroupId<(uk|us|au|eu|int|nz|ca)>/thank-you           controllers.Application.productCheckoutRouter(countryGroupId: String)
 
+GET  /$countryGroupId<(uk)>/guardian-light                         controllers.Application.productCheckoutRouter(countryGroupId: String)
+GET /us/guardian-light                                             controllers.Application.contributeGeoRedirect(campaignCode = "")
```
Comment (andrewHEguardian) [inline support-frontend/conf/routes:68]: Could you combine the redirect countries into one line?

**Review by andrewHEguardian — APPROVED** [review]
Makes sense 👍  nice and simple

File: support-frontend/assets/pages/[countryGroupId]/router.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { setUpTrackingAndConsents } from 'helpers/page/page';
 import { renderPage } from 'helpers/rendering/render';
 import { geoIds } from 'pages/geoIdConfig';
 import { Checkout } from './checkout';
+import { GuardianLightLanding } from './guardianLightLanding';
```
Comment (andrewHEguardian) [inline support-frontend/assets/pages/[countryGroupId]/router.tsx:7]: Something I realised this week - the file name was initially supposed to mirror the routes. Hence why `thank-you.ts` is spelled that way. I didn't spot this when I named the `oneTimeCheckout`, not sure if we want to use this convention?

File: support-frontend/app/controllers/Application.scala [hunk 3]
```diff
@@ -258,11 +258,20 @@ class Application(
   }
 
   def contributeGeoRedirect(campaignCode: String): Action[AnyContent] = GeoTargetedCachedAction() { implicit request =>
-    val url = List(getGeoRedirectUrl(request.geoData.countryGroup, "contribute"), campaignCode)
+    val url = getGeoPath(request, campaignCode, "contribute")
+    RedirectWithEncodedQueryString(url, request.queryString, status = FOUND)
+  }
+
+  def guardianLightGeoRedirect(campaignCode: String): Action[AnyContent] = GeoTargetedCachedAction() {
```
Comment (andrewHEguardian) [inline support-frontend/app/controllers/Application.scala:265]: Why do we support the campaignCode? What is this? It's always `= ""` in the routes file?

File: support-frontend/conf/routes (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/conf/routes:68]: We tried, there appears to be no examples of this and only of them broken out into regions. We tried `us|eu...` variants and regex but no luck.  Can revisit if we find out.

File: support-frontend/assets/pages/[countryGroupId]/router.tsx (hunk 2 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/[countryGroupId]/router.tsx:7]: Don't we have a `threeTier**Landing**` route that comes from `contribute` however? Thats what I followed here as you know its a Landing Page that way.

File: support-frontend/app/controllers/Application.scala (hunk 3 above)
Comment (paul-daniel-dempsey) [reply support-frontend/app/controllers/Application.scala]: Its not need in ths redirect so have removed.

File: support-frontend/assets/pages/[countryGroupId]/router.tsx (hunk 2 above)
Comment (andrewHEguardian) [reply support-frontend/assets/pages/[countryGroupId]/router.tsx:7]: I agree that's a better pattern as you get more from the file name than the whole file path route. Just something to think about!

**Pull request merged**